app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -0,0 +1,376 @@
+     */
+    public DatabaseAlertRepository(final AlertQueryGenerator alertQueryGenerator) {
+        _alertQueryGenerator = alertQueryGenerator;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void open() {
+        assertIsOpen(false);
+        LOGGER.debug().setMessage("Opening alert repository").log();
+        _isOpen.set(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void close() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Closing alert repository").log();
+        _isOpen.set(false);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<Alert> get(final UUID identifier) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Getting alert")
+                .addData("alertId", identifier)
+                .log();
+
+        final models.ebean.Alert ebeanAlert = Ebean.find(models.ebean.Alert.class)
+                .where()
+                .eq("uuid", identifier)
+                .findUnique();
+        if (ebeanAlert == null) {
+            return Optional.empty();
+        }
+        final Alert result = new DefaultAlert.Builder()
+                .setId(ebeanAlert.getUuid())
+                .setName(ebeanAlert.getName())
+                .setContext(ebeanAlert.getContext())
+                .setCluster(ebeanAlert.getCluster())
+                .setService(ebeanAlert.getService())
+                .setMetric(ebeanAlert.getMetric())
+                .setStatistic(ebeanAlert.getStatistic())
+                .setPeriod(Period.seconds(ebeanAlert.getPeriod()))
+                .setOperator(ebeanAlert.getOperator())
+                .setValue(new DefaultQuantity.Builder()
+                        .setValue(ebeanAlert.getQuantityValue())
+                        .setUnit(ebeanAlert.getQuantityUnit())
+                        .build())
+                .setExtensions(mergeExtensions(ebeanAlert.getNagiosExtension()))
+                .build();
+        return Optional.of(result);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public AlertQuery createQuery() {
+        assertIsOpen();
+        LOGGER.debug().setMessage("Preparing query").log();
+        return new DefaultAlertQuery(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public QueryResult<Alert> query(final AlertQuery query) {
+        assertIsOpen();
+        LOGGER.debug()
+                .setMessage("Querying")
+                .addData("query", query)
+                .log();
+
+        // Create the base query
+        final PagedList<models.ebean.Alert> pagedAlerts = _alertQueryGenerator.createAlertQuery(query);
+
+        // Compute the etag
+        // NOTE: Another way to do this would be to use the version field and hash those together.
+        final String etag = Long.toHexString(pagedAlerts.getList().stream()
```

Comment:
So, this isn't going to work like you think it is.  Unfortunately, the place that you took this from is also broken.  The gist of the problem is that the etag needs to represent the entire state of the resource, in this case that's the whole table.  Lets talk about alternatives.